`std::wstring` argument order

I now see 

```
../src/libutil/windows/environment-variables.cc:17:24: warning: narrowing conversion of ‘bufferSize’ from ‘DWORD’ {aka ‘long unsigned int’} to ‘wchar_t’ [-Wnarrowing]
   17 |     std::wstring value{bufferSize, L'\0'};
```

in Windows builds, but in 355f08a728f73f21e287926c3fded19ed975e40d we explicitly flipped it to fix a Clang MinGW warning.

Does the right argument order depend on the C++ standard library implementation? What is going on!

CC @puffnfresh

--- a/src/libutil/windows/environment-variables.cc
+++ b/src/libutil/windows/environment-variables.cc
@@ -13,8 +13,10 @@ std::optional<OsString> getEnvOs(const OsString & key)
         return std::nullopt;
     }
 
-    // Allocate a buffer to hold the environment variable value
-    std::wstring value{bufferSize, L'\0'};
+    /* Allocate a buffer to hold the environment variable value.
+       WARNING: Do not even think about using uniform initialization here,
+       we DONT want to call the initializer list ctor accidentally. */
+    std::wstring value(bufferSize, L'\0');
 
     // Retrieve the environment variable value
     DWORD resultSize = GetEnvironmentVariableW(key.c_str(), &value[0], bufferSize);

